src/pages/Governance/Proposal/hooks/useAccountHasVoted.tsx

```diff
@@ -0,0 +1,71 @@
+import {useEffect, useState} from "react";
+
+import {getTableItem} from "../../../../api";
+import {useGetAccountResource} from "../../../../api/hooks/useGetAccountResource";
+import {useWalletContext} from "../../../../context/wallet/context";
+import {GlobalState, useGlobalState} from "../../../../GlobalState";
+
+const fetchTableItem = async (
+  pool_address: string,
+  proposal_id: string,
+  handle: string,
+  state: GlobalState,
+): Promise<boolean> => {
+  const tableItemRequest = {
+    key_type: "0x1::aptos_governance::RecordKey",
+    value_type: "bool",
+    key: {
+      stake_pool: pool_address,
+      proposal_id,
+    },
+  };
+
+  await getTableItem(
+    {tableHandle: handle, data: tableItemRequest},
+    state.network_value,
+  );
+
+  return true;
+};
+
+interface VotingRecords {
+  votes: {
+    handle: string;
+  };
+}
+
+interface OwnerCapability {
+  pool_address: string;
+}
+
+export function useAccountHasVoted(proposalId: string): boolean {
+  const [state, _setState] = useGlobalState();
+  const {accountAddress} = useWalletContext();
+  const [hasVoted, setHadVoted] = useState<boolean>(false);
+
+  const {accountResource: ownerCapabilityResource} = useGetAccountResource(
+    accountAddress || "0x1",
```

Comment:
Why default to 0x1 here? 0x1 will never hold OwnerCapability

Reply:
its something I am working on, accountAddress might be null and the request function expects a string type (nut null)